src/class/CgiManager/CgiManager.cpp

```diff
@@ -99,47 +136,92 @@ void CgiManager::_childProcess(void)
 	exit(_exit_status);
 }
 
-std::string CgiManager::_parentProcess(int pid)
+void CgiManager::writeToCgi(void)
 {
-	_pipe.setParent();
-	if (_basicHTTPRequest.isBody())
-		_pipe.write(_basicHTTPRequest.getBody());
-	_pipe.closeInput();
-	return (_readCgiOutput(pid));
+	// int byte_write = write(this->getWriteFd(), this->_basicHTTPRequest.getBody().c_str(),
+	// 						   this->_basicHTTPRequest.getBody().length());
+	int byte_write = send(this->getWriteFd(), this->_basicHTTPRequest.getBody().c_str(),
+						  this->_basicHTTPRequest.getBody().length(), MSG_DONTWAIT);
```

Comment:
will we always start writing from the `this->_basicHTTPRequest.getBody().c_str()`? don't we change that by `_byte_write`?